source/extensions/filters/http/adaptive_concurrency/controller/gradient_controller.cc

```diff
@@ -215,10 +215,12 @@ void GradientController::recordLatencySample(MonotonicTime rq_send_time) {
     sample_count = hist_sample_count(latency_sample_hist_.get());
   }
 
-  if (inMinRTTSamplingWindow() && sample_count >= config_.minRTTAggregateRequestCount()) {
+  const auto agg_count = config_.minRTTAggregateRequestCount();
```

Comment:
Would it be possible to get some tests that cover this change?

Reply:
Yes, that's a fair ask, let me see what primitives other tests have to coordinate this.